src/Command/CreateSnapshotsCommand.php

```diff
@@ -59,12 +59,22 @@ public function configure()
     {
         $this->setDescription('Create a snapshots of all pages available');
         $this->addOption('site', null, InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY, 'Site id');
+        //NEXT_MAJOR: Remove the "base-console" option.
         $this->addOption('base-console', null, InputOption::VALUE_OPTIONAL, 'Base symfony console command', 'php app/console');
+        //NEXT_MAJOR: Remove the "mode" option.
         $this->addOption('mode', null, InputOption::VALUE_OPTIONAL, 'Run the command asynchronously', 'sync');
     }
 
     public function execute(InputInterface $input, OutputInterface $output)
     {
+        //NEXT_MAJOR: Remove this condition.
+        if ('php app/console' !== $input->getOption('base-console')) {
+            @trigger_error(
+                'The "base-console" is deprecated since sonata-project/page-bundle 3.27.0 and will be removed in 4.0',
+                \E_USER_DEPRECATED
+            );
+        }
+
         $siteOption = $input->getOption('site');
 
         if ([] === $siteOption) {
```

Comment:
I think it could be interesting to consider that not providing a site means doing it on every site.

This way
`[] => findAll`
`[1, 2, 3] => findByIds`

This way it reduce the logic to find the sites, and we can use directly the siteManager rather than introducing a new service.

Reply:
Hey @VincentLanglet 
Well it was the first approach that I had implemented, But it was the argument to keep the `--site=all` argument

https://github.com/sonata-project/SonataPageBundle/pull/1418#discussion_r912350492

Reply:
Didn't see this.

IMHO a command without argument can work.
And here the site is not an argument but an option, so it doesn't need to be mandatory.

But I don't know the impact of this command
- If running this command is risky, I can understand we prefer to force the list of sites.
- If running this command on all the sites, is not something risky, let's just allow it.

In the example of `doctrine:schema:update`, updating the schema is risky, so the `--force` argument makes sens.

Reply:
After we decide the final solution I will remove --site='all' from create snapshot too ;)

Reply:
> Didn't see this.
> 
> IMHO a command without argument can work. And here the site is not an argument but an option, so it doesn't need to be mandatory.
> 
> But I don't know the impact of this command
> 
> * If running this command is risky, I can understand we prefer to force the list of sites.
> * If running this command on all the sites, is not something risky, let's just allow it.
> 
> In the example of `doctrine:schema:update`, updating the schema is risky, so the `--force` argument makes sense.

Well the `doctrine:schema:update` should be a command to run just locally not in prod, the `create snapshot` or `cleanup snapshot` are commands to run everywhere, then if you go in prod and run `create:snapshot`, it's not an accident, you went there and run this manually, there isn't other command or solution to do this in production ....

well I vote to run `create:snapshot` without `--site=all` (only numbers are possible to pass as optional --site=1 --site=2)✋🏼

Reply:
I meant if we decide that
```
create:snapshot
```
will run on every sites.

It will run https://github.com/sonata-project/SonataPageBundle/blob/89013450f4e78cc79232d386ed27b9d0870c3030/src/Service/CreateSnapshotService.php#L59-L73 on every page of every site, and dunno if it can lead to weird/unwanted behavior, or if we don't care if snapshot are created/enabled and the purpose of the site id was just to have faster commands.

Reply:
Well IMO it won't have a `weird/unwanted behaviour`, because it'll have the same behaviour of `create:snapshot --site=all`, we can add the default parameter of site = `['all']`, and it will works like before.

Reply:
> Ok, so it seems natural to make optional the site option
> 
> so
> 
> ```
> create:snapshot
> ```
> 
> will be the same than
> 
> ```
> create:snapshot --site=all
> ```
> 
> And we deprecate using `--site=all`.
> 
> In next major it will be simpler if ([] === $siteIds) findAll else findByIds

agreed ;)